mbuild/tests/test_charmm_writer.py

```diff
@@ -0,0 +1,1744 @@
+                                                                                               +'/xml/'+'oplsaa.xml'},
+                                                            residues=[EthaneGOMC.name],
+                                                            reorder_res_in_pdb_psf=False,
+                                                            box=[4,5,6],
+                                                            boxes_for_simulation=1
+                                                            )
+        assert str(Test_value_0) == "<Structure 8 atoms; 1 residues; 7 bonds; PBC (orthogonal); parametrized>"
+        assert Test_value_1 == {'ETH': 0.5}
+        assert Test_value_2 == {'ETH': 0.5}
+        assert Test_value_3 ==  ['ETH']
+
+
+    def test_Specific_FF_to_no_atoms_in_residue(self):
+        Empty_compound = mb.Compound()
+
+        Test_value_0, Test_value_1, \
+        Test_value_2, Test_value_3 = specific_ff_to_residue(Empty_compound,
+                                                            forcefield_selection={'Empty_compound': 'oplsaa'},
+                                                            residues=[],
+                                                            reorder_res_in_pdb_psf=False,
+                                                            box=[5, 6, 7],
+                                                            boxes_for_simulation=1
+                                                            )
+        print('Test_value_0 = ' +str(Test_value_0))
+        assert Test_value_0 is None
+        assert Test_value_1 is None
+        assert Test_value_2 is None
+        assert Test_value_3 is None
+
+
+
+
+    def test_charmm_correct_residue_format(self, EthaneGOMC):
+        try:
+            Test_value = Charmm(EthaneGOMC, 'box_0',
+                                structure_box_1 = None,
+                                filename_box_1 = None,
+                                FF_filename=None,
+                                residues = [EthaneGOMC.name],
+                                forcefield_selection = {EthaneGOMC.name : 'oplsaa'},
+                                )
+
+        except:
+            Test_value = "TEST_FAILED"
+
+        assert Test_value.input_error is False
+
+    def test_charmm_residue_not_None(self, EthaneGOMC):
+        try:
+            Test_value = Charmm(EthaneGOMC, 'box_0',
+                                structure_box_1=None,
+                                filename_box_1=None,
+                                FF_filename=None,
+                                residues=EthaneGOMC.name,
+                                forcefield_selection={EthaneGOMC.name: 'oplsaa'}
+                                )
+        except:
+            Test_value = "TEST_FAILED"
+
+        assert Test_value == "TEST_FAILED"
+
+    def test_charmm_residue_string(self, EthaneGOMC):
+        try:
```

Comment:
I could be wrong, but I'm betting you can use `pytest.raises(ErrorType)` here.

E.g., say I have a test that I know will fail. When I call `function` with `arguments` the code is going to raise a `ValueError` with the phrase `Missing residue ...`. I can write a test with:

```python

with pytest.raises(ValueError, match="Missing residue"):
    function_with_problem(arguments)

```

Reply:
All the test files were changed from a general failure check to the error specific failure method with the pytest.raises() function